### ISSUE TITLE
musl support

### ISSUE THREAD

**Issue by jbboehr** [noporpoise/sort_r#12]
So I'm using this repo and it's great! But I tried compiling my project on alpine linux recently, which uses musl, which *apparently* doesn't provide an implementation of `qsort_r`. They also don't define any macros AFAIK that let you detect musl.

I'll see if I can think of a nice way to fix it, but it seems like it might just require a flag. I would just `#undef` some macros to hack it in but none of these look terribly safe to `#undef`:

```c
#if (defined __APPLE__ || defined __MACH__ || defined __DARWIN__ || \
     defined __FreeBSD__ || defined __DragonFly__)
#  define _SORT_R_BSD
#elif (defined _GNU_SOURCE || defined __gnu_hurd__ || defined __GNU__ || \
       defined __linux__ || defined __MINGW32__ || defined __GLIBC__)
#  define _SORT_R_LINUX
#elif (defined _WIN32 || defined _WIN64 || defined __WINDOWS__)
#  define _SORT_R_WINDOWS
#  undef _SORT_R_INLINE
#  define _SORT_R_INLINE __inline
#else
  /* Using our own recursive quicksort sort_r_simple() */
#endif
```

**Comment by jbboehr** [comment]
Well, I'm dumb. Looks like you export sort_r_simple anyway. I'll just add a check to autoconf and:

```c
#ifndef HAVE_QSORT_R
#define sort_r sort_r_simple
#endif
```

I'll leave this open for now in case someone wants to come up with a better solution.

**Comment by filoozom** [comment]
Just had the same issue. Wouldn't another solution be to change this:

https://github.com/noporpoise/sort_r/blob/c8c65c1e183df8f3bd1b221e27653d9705ff4abe/sort_r.h#L281

to:

```c
#if !defined __GLIBC__ || ((__GLIBC__ < 2) || (__GLIBC__ == 2 && __GLIBC_MINOR__ < 8)) 
```

This way, we only use `qsort_r` if it's a known supported version of `glibc`. Of course, this might break other systems I'm not aware of, which is why I'm asking the question.

**Comment by jbboehr** [comment]
@filoozom It's working for me on alpine!

**Comment by noporpoise** [comment]
I've proposed a fix with PR #18. I'll merge if it looks good.

**Comment by noporpoise** [comment]
Thanks for raising this issue - please let me know if the latest version works ok.
